clang/include/clang/Sema/Sema.h

```diff
@@ -9219,7 +9222,8 @@ class Sema final : public SemaBase {
   /// \returns true if an error occurred, false otherwise.
   bool CheckTemplateArgumentList(
       TemplateDecl *Template, SourceLocation TemplateLoc,
-      TemplateArgumentListInfo &TemplateArgs, bool PartialTemplateArgs,
+      TemplateArgumentListInfo &TemplateArgs,
```

Comment:
I wonder if instead we want to add an additional method.
In a lot of places through the rest of the patch, `DefaultArgs` is defaulted

Reply:
I am not seeing a worthwhile tradeoff.

Just the function signature is hugely complicated, and would need to be duplicated.
The implementation would be mostly the same, with a small block which would be omitted in one of the implementations.

What did you have in mind?

Reply:
Yeah. One issue I have often had with these functions with large amount of both defaulted and non-defaulted parameters, is that you would want to extend it by changing the signature, then arguments would match parameters incorrectly, but this would not cause a hard error on all of the call sites.

I could have easily added DefaultArgs as defaulted empty here, but chose not to due to this reason.

Besides that, overloading functions with such huge numbers of parameters creates some confusion as well.
I'd slightly prefer if we avoided that, but don't have strong enough feelings to go on a crusade against it.

Reply:
Can we think about ways to simplify these interface in a subsequent patch?
@AaronBallman @erichkeane for additional opinions

Reply:
The problem with the separate Sugared / Converted lists I already have an incomplete patch to address that, though  I am currently working on something else.